systems/billing/collector/internal/server/event.go

```diff
@@ -0,0 +1,280 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	operatorpb "github.com/ukama/telna/cdr/pb/gen"
+	client "github.com/ukama/ukama/systems/billing/collector/internal/clients"
+	epb "github.com/ukama/ukama/systems/common/pb/gen/events"
+	subpb "github.com/ukama/ukama/systems/subscriber/registry/pb/gen"
+	simpb "github.com/ukama/ukama/systems/subscriber/sim-manager/pb/gen"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// TODO: We need to think about retry policies for failing interaction between our backend and the upstream billing service
+// provider
+
+const (
+	handlerTimeoutFactor = 3
+)
+
+type BillingCollectorEventServer struct {
+	client client.BillingClient
+	epb.UnimplementedEventNotificationServiceServer
+}
+
+func NewBillingCollectorEventServer(client client.BillingClient) *BillingCollectorEventServer {
+	return &BillingCollectorEventServer{
+		client: client,
+	}
+}
+
+func (b *BillingCollectorEventServer) EventNotification(ctx context.Context, e *epb.Event) (*epb.EventResponse, error) {
+	log.Infof("Received a message with Routing key %s and Message %+v", e.RoutingKey, e.Msg)
+
+	switch e.RoutingKey {
+
+	// Send usage event
+	case "event.cloud.cdr.sim.usage":
+		msg, err := unmarshalOperatorSimUsage(e.Msg)
+		if err != nil {
+			return nil, err
+		}
+
+		err = handleCdrSimUsageEvent(e.RoutingKey, msg, b)
+		if err != nil {
+			return nil, err
+		}
+
+	// Create customer
+	case "event.cloud.registry.subscriber.create":
+		msg, err := unmarshalSubscriber(e.Msg)
+		if err != nil {
+			return nil, err
+		}
+
+		err = handleRegistrySubscriberCreateEvent(e.RoutingKey, msg, b)
+		if err != nil {
+			return nil, err
+		}
+
+	// Update customer
+	case "event.cloud.registry.subscriber.update":
+		msg, err := unmarshalSubscriber(e.Msg)
+		if err != nil {
+			return nil, err
+		}
+
+		err = handleRegistrySubscriberUpdateEvent(e.RoutingKey, msg, b)
+		if err != nil {
+			return nil, err
+		}
+
+	// Delete customer
+	case "event.cloud.registry.subscriber.delete":
```

Comment:
It's kind of confusing I think we need to add system name to events

Reply:
> It's kind of confusing I think we need to add system name to events

I think the confusion comes from the fact that the **subscriber-registry** service (inside the **subscriber system)** was renamed `registry`, when we already have a system that is also called `registry`.